### PR TITLE
Add logic to filter Freaks

### PR DESCRIPTION
**Pull request by DejanuTeodora**
## Description
Added logic to filter freaks by skills.

## Related
https://trello.com/c/9v3osWKH

## Steps to Test or Reproduce
1. Install app
2. Open app
3. Tap on skills button 
4. Select skills
5. Tap on apply button

### CODE REVIEW

**Comment by DejanuTeodora** [comment]
[iOS - Skills Filter Screen - Add Apply Button](https://trello.com/c/9v3osWKH/106-ios-skills-filter-screen-add-apply-button)

File: Shared/Features/Freaks/View/SkillsView.swift [hunk 1]
```diff
@@ -6,12 +6,26 @@ struct SkillsView: View {
...
+                    ForEach(viewModel.skills, id: \.self) { skill in
+                        MultipleSelectionRow(viewModel: viewModel, skill: skill, isSelected: viewModel.isSkillSelected(skill: skill)) {
+                            viewModel.setupSelectedSkills(skill: skill)
+                        }
                     }
                 }
+
+                Button(action: {
+                    self.showSkillsView = false
+                    self.viewModel.isFilterApplied = true
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/View/SkillsView.swift:20]: We also need to reset isFilterApplied somewhere.

File: Shared/Features/Freaks/ViewModel/FreaksViewModel.swift [hunk 2]
```diff
@@ -1,10 +1,28 @@
 import Foundation
+import UIKit
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/ViewModel/FreaksViewModel.swift:2]: Why are we importing UIKit here?

**Review by vladstanescu94 — CHANGES_REQUESTED** [review]

File: Shared/Features/Freaks/View/SkillsView.swift [hunk 3]
```diff
@@ -6,16 +6,31 @@ struct SkillsView: View {
...
+                    .foregroundColor(Color("SecondaryColor"))
+                    .frame(maxWidth: 150, maxHeight: 40)
+                    .background(Color("AccentColor"))
+                    .padding()
+                    .font(.system(size: 18, weight: .bold, design: .default))
             }
             .navigationBarTitle(Text("SKILLS"), displayMode: .inline)
             .navigationBarItems(trailing: Button(action: {
                 viewModel.skillSelections.removeAll()
+                viewModel.isFilterApplied = false
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/View/SkillsView.swift:33]: Now we are resetting this var if we tap on the Reset button, but what happens if we just close the sheet or press the X button?

**Review by vladstanescu94 — APPROVED** [review]

**Review by RalucaMihaela — APPROVED** [review]

**Pull request merged**